File: Main (2).java
import java.util.Scanner;

public class Main {
    public static int reverseNumber(int num) {
        int reversed = 0;
        while (num > 0) {
            reversed = reversed * 10 + num % 10;
            num /= 10;
        }
        return reversed;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt(); // Вводим количество чисел
        int prev = sc.nextInt(); // Вводим первое число

        StringBuilder result = new StringBuilder();

        for (int i = 1; i < n; i++) {
            int current = sc.nextInt(); // Вводим следующее число
            if (current < prev) {
                result.insert(0, reverseNumber(prev) + " ");
            }
            prev = current;
        }

        result.insert(0, reverseNumber(prev) + " "); // Добавляем последнее число
        System.out.println(result.toString().trim()); // Переворачиваем и выводим результат
    }
}

File: main (2).rb
def reverse_number(num)
  num.to_s.reverse.to_i
end

n = gets.to_i
prev = gets.to_i
result = []

(n-1).times do
  current = gets.to_i
  if current < prev
    result.unshift(reverse_number(prev)) # Добавляем в начало
  end
  prev = current
end

result.unshift(reverse_number(prev)) # Добавляем последнее число

# Переворачиваем и выводим
puts result.reverse.join(" ")
